### ISSUE TITLE
Clippy complains about `drop(&value)`

### ISSUE THREAD

**Issue by novacrazy** [kang-sw/capture-it#1]
`cargo clippy`
```rust
error: calls to `std::mem::drop` with a reference instead of an owned value. Dropping a reference does nothing
   --> src\main.rs:1:19
  |
1 |       let mut gen = capture!([*index = 0], move || {
  |  ___________________^
2 | |         index += 1;
3 | |         index
4 | |     });
  | |______^
```
```
calls to `std::mem::drop` with a reference instead of an owned value. Dropping a reference does nothing
for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#drop_ref
`#[deny(clippy::drop_ref)]` on by default
```

Could add `#[allow(clippy::drop_ref)]` before every drop, or modify the macro to do drops in a DRY manner. 

Alternatively, you could use variable shadowing to explicitly capture it:
```diff
- drop(&value);
+ let mut value = value;
```

The latter is probably more reliable.

**Comment by kang-sw** [comment]
Oh, thanks for reporting the issue. Didn't think to run clippy, will check and incorporate, thanks!

**Comment by kang-sw** [comment]
Fixed by e8ae55fc18159d2693fa826c2b59355af0b015f5, This was solved by replacing the `drop(...` expression with `let _ = &...`. (It seems that something like `let mut ... = ...`, which you suggested, would pass ownership of the captured variable to the local variable, so that all closures would be recognised as `FnOnce`)

Again, thanks for spotting the issue!

**Comment by novacrazy** [comment]
Ah, makes sense. Thanks!
